contrib/cell_star/core/seeder.py

```diff
@@ -0,0 +1,309 @@
+# -*- coding: utf-8 -*-
+__author__ = 'Adam Kaczmarek, Filip Mróz'
+
+# External imports
+import random
+from copy import copy
+# Internal imports
+# Objects
+from contrib.cell_star.core.seed import Seed
+# Utils
+from contrib.cell_star.utils.cluster_util import *
+from contrib.cell_star.utils.calc_util import *
+from contrib.cell_star.utils.image_util import *
+
+
+class Seeder(object):
+
+    def __init__(self, images, parameters):
+        self.images = images
+        random.seed(abs(np.sum(images.image)))
+        self.parameters = parameters
+        self.cluster_min_distance = self.parameters["segmentation"]["seeding"]["minDistance"] \
+            * self.parameters["segmentation"]["avgCellDiameter"]
+        self.random_radius = self.parameters["segmentation"]["seeding"]["randomDiskRadius"] \
+            * self.parameters["segmentation"]["avgCellDiameter"]
+
```

Comment:
I entered this routine while segmenting a sample yeast image with ~1500 seeds, all of which were initially less than the cluster_min_distance from some adjacent seed. I am guessing that the internal loop is going to run at least 750 times before it converges on a cluster. Ideally, a Delaunay Triangulation could be used to get the adjacent neighbors of a seed and a merge could be done just considering those neighbors and could be done in-place on that data structure. That would be fastest since only a few minimum distances would need to be recomputed instead of 1500 * 2 per step and the memory moving (in this case, ~ 1Mb per iteration) required by the delete step on line 55 could be avoided.
An intermediate approach is to process the array of 1500 at once, coalescing all seed pairs that are closest to each other and less than the cluster_min_distance apart from each other. That can all be done in Numpy and hopefully the first iteration of the loop. In my case, 400 pairs are closest to each other (i is closest to j and j is closest to i), reducing the array on the first step to 1100 cases. Hopefully, the clustering should be complete after a few iterations and I am pretty sure you are guaranteed to converge.